lib/src/main/java/io/ably/lib/types/Annotation.java

```diff
@@ -0,0 +1,248 @@
+package io.ably.lib.types;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import io.ably.lib.util.Log;
+import io.ably.lib.util.Serialisation;
+import org.msgpack.core.MessageFormat;
+import org.msgpack.core.MessagePacker;
+import org.msgpack.core.MessageUnpacker;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+public class Annotation extends BaseMessage {
+
+    private static final String TAG = Annotation.class.getName();
+
+    private static final String ACTION = "action";
+    private static final String SERIAL = "serial";
+    private static final String MESSAGE_SERIAL = "messageSerial";
+    private static final String TYPE = "type";
+    private static final String NAME = "name";
+    private static final String COUNT = "count";
+    private static final String EXTRAS = "extras";
+
+    /**
+     * (TAN2b) The action, whether this is an annotation being added or removed,
+     * one of the AnnotationAction enum values.
+     */
+    public AnnotationAction action;
+
+    /**
+     * (TAN2i) This annotation's unique serial (lexicographically totally ordered).
+     */
+    public String serial;
```

Comment:
```suggestion
    public String messageSerial;
```
https://github.com/ably/specification/pull/292#discussion_r2135598426

Reply:
This is more self-explanatory. We can update the serialization key accordingly.